From be8e9ced5ffbb3474c5bf679150adcd3c6737dc2 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Fri, 13 Jan 2023 10:48:06 +0200
Subject: [PATCH 38/63] mailbox: llce: Mark all LLCE packages as long messages

This reflects the LLCE FW restrictions. Otherwise, the data received on
LLCE logger interfaces might get corrupted.

Issue: ALB-9261
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/mailbox/llce-mailbox.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mailbox/llce-mailbox.c b/drivers/mailbox/llce-mailbox.c
index 80e435a763a3..6f0794b891ce 100644
--- a/drivers/mailbox/llce-mailbox.c
+++ b/drivers/mailbox/llce-mailbox.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause
 /*
- * Copyright 2020-2022 NXP
+ * Copyright 2020-2023 NXP
  */
 #include <dt-bindings/mailbox/nxp-llce-mb.h>
 #include <linux/can/dev.h>
@@ -1372,6 +1372,9 @@ static int process_pop_logger(struct mbox_chan *chan, struct llce_rx_msg *msg)
 	struct llce_can_mb *frame;
 	bool pop;
 
+	/* Logger works with long MBs only */
+	msg->rx_pop.mb.is_long = true;
+
 	/* Use a stashed frame */
 	pop = pop_llce_logger_data(chan, &frame, &mb_index);
 	if (pop) {
-- 
2.25.1

